src/kernel/src/process/mod.rs

```diff
@@ -409,6 +416,26 @@ impl VProc {
         Ok(SysOut::ZERO)
     }
 
+    fn sys_thr_set_name(self: &Arc<Self>, i: &SysIn) -> Result<SysOut, SysErr> {
+        let name: &str = unsafe { i.args[0].to_str(32) }?.unwrap();
```

Comment:
The PS4 will use an empty name if this argument is null.

Reply:
Fixed., thanks.